internal/services/netapp/netapp_volume_quota_rule_resource.go

```diff
@@ -0,0 +1,314 @@
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.PoolName,
+		},
+
+		"volume_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.VolumeName,
+		},
+
+		"quota_target": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"quota_size_in_kib": {
+			Type:         pluginsdk.TypeInt,
+			Optional:     true,
+			ValidateFunc: validation.IntAtLeast(4),
+		},
+
+		"quota_type": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringInSlice(volumequotarules.PossibleValuesForType(), false),
+		},
+	}
+}
+
+func (r NetAppVolumeQuotaRuleResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r NetAppVolumeQuotaRuleResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 90 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.VolumeQuotaRules
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model netAppModels.NetAppVolumeQuotaRuleModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := volumequotarules.NewVolumeQuotaRuleID(subscriptionId, model.ResourceGroupName, model.AccountName, model.CapacityPoolName, model.VolumeName, model.Name)
+			volumeID := volumes.NewVolumeID(subscriptionId, model.ResourceGroupName, model.AccountName, model.CapacityPoolName, model.VolumeName)
+
+			metadata.Logger.Infof("Import check for %s", id)
+			existing, err := client.Get(ctx, id)
+			if err != nil && existing.HttpResponse.StatusCode != http.StatusNotFound {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			// Performing some validations that are not possible in the schema
+			if errorList := netAppValidate.ValidateNetAppVolumeQuotaRule(ctx, volumeID, metadata.Client, pointer.To(model)); len(errorList) > 0 {
+				return fmt.Errorf("one or more issues found while performing deeper validations for %s:\n%+v", id, errorList)
+			}
+
+			parameters := volumequotarules.VolumeQuotaRule{
+				Location: location.Normalize(model.Location),
+				Properties: &volumequotarules.VolumeQuotaRulesProperties{
+					QuotaSizeInKiBs: pointer.To(model.QuotaSizeInKiB),
+					QuotaType:       pointer.To(volumequotarules.Type(model.QuotaType)),
+					QuotaTarget:     utils.String(model.QuotaTarget),
+				},
+			}
+
+			err = client.CreateThenPoll(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			// Waiting for volume quota rule to be completely provisioned
+			if err := waitForVolumeQuotaRuleCreateOrUpdate(ctx, client, id); err != nil {
```

Comment:
Is this needed? `CreateThenPoll` shouldn't return until the resource is finished provisioning.

Reply:
This was failing misereably without the wait function when creating 5 rules at the same time, so I suggest we keep this.